src/Prelude/Linear.hs

```diff
@@ -74,3 +83,29 @@ maybe _ f (Just y) = f y
 -- arrow but we have a linear arrow
 forget :: (a ->. b) ->. a -> b
 forget f x = f x
+
+-- | Linearly typed replacement for the standard 'foldr' function, to allow
```

Comment:
This `'foldr'` will point to this same function. Use `'Prelude.foldr'` for correct hyperlinks. I think.